android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt

```diff
@@ -142,6 +151,19 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
     return (filter == null || filter.equals(this.idElement.idPart))
   }
 
+  protected suspend fun Patient.canAddOverdue(showOverdue: Boolean): Boolean {
```

Comment:
what does this function do? not sure I understand

Reply:
actually, this function is checking that patient would be shown or not in the patient list like if the `showOverdue` param will `true` then we will show all the patient in the list either the patient second vaccination date is overdue or not and if the `showOverdue` param will `false` then we will hide all those patient who have not received vaccination after the specific time.

Reply:
still not sure I follow, if it's checking status, can't this work https://github.com/opensrp/fhircore/blob/d4dd6042be7c4594aee4213d7a36e165219e90d6/android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt#L113?

Reply:
could you elaborate a bit more on what's not working?

Reply:
A filter needs to be applied to display only patients with overdue vaccines  when the `Show overdue` button is toggled on. Logic to filter out overdue patients is already implemented https://github.com/opensrp/fhircore/pull/194#discussion_r645452420. Why not use that when filtering here?

Reply:
When the button is toggled off, we should display all patients - both overdue and not.

Reply:
fixed.